### DISCUSSION TITLE
ERR: no one is publishing to path 'xxxxx'

### DISCUSSION THREAD

**Discussion by Diddlik** [bluenviron/mediamtx#432] [Questions]
## Which version are you using?

v0.16.2

## Which operating system are you using?

OS

- [x] Linux
- [x] Docker

Architecture

- [x] amd64

## Describe the issue

I have a stream from the netatmo camera (http://192.168.1.22/xxxxxxxxxxxxxxxxxxx/live/files/high/index.m3u8). This is read out with ffmpeg. 
I have tried both Docker and Ubuntu. In both cases, after a certain time, the stream is no longer transmitted and the error message appears. 
The clients are: Surveillance Station from Synology and Home Assistant.
Restarting the rtsp-simple-server only helps for a short time. 

Config:

```
 paths:
      cam:
       runOnInit: ffmpeg -i http://192.168.1.22/xxxxxxxxxxxxxxxxxxx/live/files/high/index.m3u8 -c:v copy -f rtsp tsp://localhost:$RTSP_PORT/$RTSP_PATH
       runOnInitRestart: yes

```

Log from server. 

```Jun 04 07:23:42 york rtsp-simple-server[4170081]: 2021/06/04 07:23:42 I [0/0] [RTSP] [conn 192.168.1.3:44862] opened
Jun 04 07:23:42 york rtsp-simple-server[4170081]: 2021/06/04 07:23:42 I [0/0] [RTSP] [conn 192.168.1.3:44862] ERR: invalid URL (/cam)
Jun 04 07:23:42 york rtsp-simple-server[4170081]: 2021/06/04 07:23:42 I [0/0] [RTSP] [conn 192.168.1.3:44862] closed
Jun 04 07:23:42 york rtsp-simple-server[310398]: [214B blob data]
Jun 04 07:23:43 york rtsp-simple-server[4170081]: 2021/06/04 07:23:43 I [0/0] [RTSP] [conn 192.168.1.3:44866] opened
Jun 04 07:23:43 york rtsp-simple-server[4170081]: 2021/06/04 07:23:43 I [0/0] [RTSP] [conn 192.168.1.3:44866] ERR: no one is publishi
Jun 04 07:23:43 york rtsp-simple-server[4170081]: 2021/06/04 07:23:43 I [0/0] [RTSP] [conn 192.168.1.3:44866] closed
Jun 04 07:23:43 york rtsp-simple-server[4170081]: 2021/06/04 07:23:43 I [0/0] [RTSP] [conn 192.168.1.3:44870] opened
Jun 04 07:23:43 york rtsp-simple-server[4170081]: 2021/06/04 07:23:43 I [0/0] [RTSP] [conn 192.168.1.3:44870] ERR: no one is publishi
Jun 04 07:23:43 york rtsp-simple-server[4170081]: 2021/06/04 07:23:43 I [0/0] [RTSP] [conn 192.168.1.3:44870] closed
```

**Comment by aler9** [comment]
Hello, in my opinion FFmpeg is not receiving and / or converting the stream from HLS to RTSP correctly.

This can be caused by different reasons:

* maybe the stream can't be received correctly
* maybe the stream can't be remuxed into RTSP

In my opinion, you have to:

* try ffmpeg manually without using `runOnInit` and check whether there's any error
* try disabling the audio track; from my experience, HLS audio codecs are often incompatible with RTSP:
  ```
  runOnInit: ffmpeg -i http://192.168.1.22/xxxxxxxxxxxxxxxxxxx/live/files/high/index.m3u8 -map 0:v:0 -c:v copy -f rtsp tsp://localhost:8554/cam
  ```

**Comment by sravich1** [comment]
Hi all,
getting same error ...how to solve to get rtsp streaming..kindly let me know

0:34:51.002808595] [5664]  INFO RPI vc4.cpp:437 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media3 and ISP device /dev/media0
[0:34:51.003581199] [5660]  INFO Camera camera.cpp:1033 configuring streams: (0) 1920x1080-YUV420
[0:34:51.004098997] [5664]  INFO RPI vc4.cpp:565 Sensor: /base/soc/i2c0mux/i2c@1/ov5647@36 - Selected sensor format: 1920x1080-SGBRG10_1X10 - Selected unicam format: 1920x1080-pGAA
2024/01/24 13:55:18 INF [path cam] [RPI Camera source] ready: 1 track (H264)
2024/01/24 13:57:10 INF [RTSP] [conn 10.35.62.103:54176] opened
2024/01/24 13:57:10 INF [RTSP] [conn 10.35.62.103:54176] closed: invalid URL (/h264)
2024/01/24 13:57:22 INF [RTSP] [conn 10.35.62.103:54180] opened
2024/01/24 13:57:22 INF [RTSP] [conn 10.35.62.103:54180] closed: no one is publishing to path 'h264'
^C2024/01/24 13:57:30 INF shutting down gracefully
2024/01/24 13:57:30 INF [SRT] listener is closing

I am using raspberrypi4 or compute module to run the same...
if this done..how to get ONVIF going with this
--Thanks
Ravi

**Comment by cubisttriangle** [comment]
In your post, there is a typo in what I assume should be a rtsp url. You have written `tsp://`, rather than `rtsp://`.

**Comment by aler9** [comment]
This discussion has become a collection of unrelated problems. This server works by serving different streams into different paths. The error "no one is publishing to path 'XXX'" means that you're trying to read path "XXX" but no one is publishing anything inside it.
